pkg/storage/bloom/v1/bloom_tokenizer.go

```diff
@@ -184,60 +167,55 @@ func (bt *BloomTokenizer) sendBloom(
 	bt.metrics.bloomSize.Observe(float64(bloom.ScalableBloomFilter.Capacity() / eightBits))
 	bt.metrics.bloomsTotal.Inc()
 	ch <- &BloomCreation{
-		Bloom:            bloom,
-		SourceBytesAdded: bytesAdded,
+		Bloom: bloom,
+		Stats: stats,
 	}
 }
 
-// addChunkToBloom adds the tokens from the given chunk to the given bloom.
-// It continues until the chunk is exhausted or the bloom is full.
-// NB(owen-d): We ensure the invariant that each line is indexed entirely into at least one bloom.
-// This includes both raw ngrams and chunk-prefixed ngrams and is why we use a peeking iterator --
-// so we can advance the iterator only after we're sure the bloom has accepted the line.
-// This is because the _line_ is the atom in Loki's data model and a query must either match (or not) an individual line.
-// Therefore, we index entire lines into a bloom to ensure a lookups are accurate.
-func (bt *BloomTokenizer) addChunkToBloom(bloom *Bloom, ref ChunkRef, entryIter v2iter.PeekIterator[push.Entry]) (full bool, bytesAdded int) {
+func prefixForChunkRef(chk ChunkRef) []byte {
+	enc := encoding.EncWith(make([]byte, 20))
+	enc.PutBE64(uint64(chk.From))    // 8 bytes
+	enc.PutBE64(uint64(chk.Through)) // 8 bytes
+	enc.PutBE32(chk.Checksum)        // 4 bytes
+	return enc.Get()
+}
+
+// addChunkToBloom adds the structured metadata from the given chunk to the given bloom.
+func (bt *BloomTokenizer) addChunkToBloom(bloom *Bloom, ref ChunkRef, entryIter v2iter.PeekIterator[push.Entry]) (bool, IndexingStats) {
 	var (
-		tokenBuf, prefixLn = prefixedToken(bt.lineTokenizer.N(), ref, nil)
-		tokens             int
-		successfulInserts  int
-		cachedInserts      int
-		collisionInserts   int
-		chunkBytes         int
-		linesAdded         int
+		tokens            int
+		successfulInserts int
+		cachedInserts     int
+		collisionInserts  int
+		linesAdded        int
 	)
 
+	// return values
+	full, stats := false, newIndexingStats()
+
+	tokenizer := NewStructuredMetadataTokenizer(string(prefixForChunkRef(ref)))
+
 	// We use a peeking iterator to avoid advancing the iterator until we're sure the bloom has accepted the line.
 outer:
 	for entry, ok := entryIter.Peek(); ok; entry, ok = entryIter.Peek() {
-		line := entry.Line
-		chunkBytes += len(line)
-
-		tokenItrs := []v2iter.Iterator[[]byte]{
-			// two iterators, one for the raw tokens and one for the chunk prefixed tokens.
-			// Warning: the underlying line tokenizer (used in both iterators) uses the same buffer for tokens.
-			// They are NOT SAFE for concurrent use.
-			NewPrefixedTokenIter(tokenBuf, prefixLn, bt.lineTokenizer.Tokens(line)),
-			bt.lineTokenizer.Tokens(line),
-		}
+		for _, kv := range entry.StructuredMetadata {
+			stats.SourceBytes += len(kv.Name) + len(kv.Value)
+			stats.Fields.Add(Field(kv.Name))
```

Comment:
Maybe it should not be called stats. It's metadata from the indexing.

Reply:
"Stats" also threw me off, since it makes me think stats are for observability and not for business logic. 

IndexingInfo would be a little more clear to me, personally.

Reply:
What about renaming this to `metadata` or `IndexingInfo` as Robert suggests for now?

Reply:
I prefer `IndexingInfo` over `Metadata`, since the latter is associated with Structured Metadata as well